### DISCUSSION TITLE
View collection table type

### DISCUSSION THREAD

**Discussion by gaganbiswas** [pocketbase/pocketbase#4372] [Q&A] [answered]
I am building a library application for managing books at my college. We have around 1000 books or so which are available for the students to take. 

I have two tables: 
1. `books` --> All the books that are currently in the library and their location
2. `issued_books` --> Table which keeps track of the books issued to a student. It has a `book_ids` field which contains relation to the `books` table.
3. `available_books` --> This is the one i need help for since i cannot understand the view collection. I want this to have all the books from the `books` table which are not a part of `book_ids` in the `available_books` table.

The issue is I have tried doing something like:
`SELECT books.id, books.name, books.location FROM books WHERE books.id NOT IN (SELECT issued_books.book_ids FROM issued_books);`

But this is not working for me. Can you point me out what I am doing wrong and suggest me something.
P.S. Not so good in SQL so nevermind some mistakes i may have made in the above.

**Comment by ganigeorgiev** [comment, marked as answer]
Multi-valued fields are stored as serialized json array so you'll have to use the SQLite [`json_each`](https://www.sqlite.org/json1.html#jeach) function if you want to extract or apply a constraint against each value of the array column.

There are several different ways you can structure the query. One possible example could be something like (_I haven't tested it_):

```sql
SELECT
   b.id,
   b.name,
   b.location
FROM books b
WHERE NOT EXISTS (
    SELECT 1
    FROM issued_books ib
    INNER JOIN json_each(ib.book_ids) je ON je.value = b.id
)
```

The main thing to note from the above is that `json_each` creates a inmemory/virtual table of the array values and `je.value` refers to a single element of the json array.

**Reply by gaganbiswas** [reply]
> Multi-valued fields are stored as serialized json array so you'll have to use the SQLite [`json_each`](https://www.sqlite.org/json1.html#jeach) function if you want to extract or apply a constraint against each value of the array column.

This was it. Thank you. I tested it and works as expected 😁
